### PR TITLE
Provide connection options to host process

### PR DESCRIPTION
**Pull request by kburtram**

### CODE REVIEW

**Comment by kburtram** [comment]
Ping?  Could someone take a look at this PR?

File: src/Microsoft.SqlTools.ServiceLayer/ServiceHost.cs [hunk 1]
```diff
@@ -171,6 +172,140 @@ public void RegisterInitializeTask(InitializeCallback callback)
...
+        internal async Task HandleCapabilitiesRequest(
+            CapabilitiesRequest initializeParams, 
+            RequestContext<CapabilitiesResult> requestContext)            
+        {
+            await requestContext.SendResult(
+                new CapabilitiesResult
+                {
+                    Capabilities = new DmpServerCapabilities
+                    {
+                        ProtocolVersion = "1.0",
```
Comment (coquagli) [inline src/Microsoft.SqlTools.ServiceLayer/ServiceHost.cs]: I would personally be in favor of making these `readonly`/`const` at the top of the file, but it's obviously not 100% necessary.

File: src/Microsoft.SqlTools.Hosting/Hosting/Contracts/ConnectionProviderOptions.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        public static readonly string SpecialValueAuthType = "authType";
+        public static readonly string SpecialValueUserName = "userName";
+        public static readonly string SpecialValuePasswordName = "password";
+
+        public string Name { get; set; }
+
+        public string DisplayName { get; set; }
+
+        /// <summary>
+        /// Type of the parameter.  Can be either string, number, or category.
```
Comment (coquagli) [inline src/Microsoft.SqlTools.Hosting/Hosting/Contracts/ConnectionProviderOptions.cs]: Should the setter enforce the allowed values?

Comment (kburtram) [inline src/Microsoft.SqlTools.Hosting/Hosting/Contracts/ConnectionProviderOptions.cs]: Good idea.  I'll add that logic in follow-up change.

**Pull request merged**